Lib/test/test_types.py

```diff
@@ -694,6 +695,15 @@ def test_or_type_operator_with_NewType(self):
     def test_or_type_operator_with_IO(self):
         assert typing.IO | str == typing.Union[typing.IO, str]
 
+    def test_or_type_operator_with_SpecialForm(self):
+        assert typing.Any | str == typing.Union[typing.Any, str]
+        assert typing.NoReturn | str == typing.Union[typing.NoReturn, str]
+        assert typing.Optional[int] | str == typing.Union[typing.Optional[int], str]
+        assert typing.Union[int, bool] | str == typing.Union[typing.Union[int, bool], str]
```

Comment:
`typing.Union` flattens its args.
```suggestion
        assert typing.Union[int, bool] | str == typing.Union[int, bool, str]
```